presto-docs/src/main/sphinx/admin/cte-materialization.rst

```diff
@@ -0,0 +1,103 @@
+    This can also be specified on a per-query basis using the ``cte_heuristic_replication_threshold`` session property.
+
+``query.cte-hash-partition-count``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+    * **Type:** ``integer``
+    * **Default value:** ``100``
+
+    The number of partitions to be used for materializing Common Table Expressions (CTEs) in queries.
+    This setting determines how many buckets or writers should be used when materializing the CTEs, potentially affecting the performance of queries involving CTE materialization.
+    A higher number of partitions might improve parallelism but also increases overhead in terms of memory and network communication.
+    Recommended value: 4 - 10x times the size of the cluster.
+    This can also be specified on a per-query basis using the ``cte_hash_partition_count`` session property.
+
+``query.cte-partitioning-provider-catalog``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+    * **Type:** ``string``
+    * **Default value:** ``system``
+
+    The name of the catalog to be used and which provides custom partitioning for Common Table Expression (CTE) materialization.
+    This setting specifies which catalog should be used for CTE materialization and for determining how to partition the materialization of CTEs in queries.
+    This can also be specified on a per-query basis using the ``cte_partitioning_provider_catalog`` session property.
+
+``cte-filter-and-projection-pushdown-enabled``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+    * **Type:** ``boolean``
+    * **Default value:** ``true``
+
+    Flag to enable or disable the pushdown of common filters and projects into the materialized CTE.
+    This can also be specified on a per-query basis using the ``cte_filter_and_projection_pushdown_enabled`` session property.
+
+``hive.temporary-table-storage-format``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+    * **Type:** ``string``
+    * **Allowed values:** ``PAGEFILE``, ``ORC``, ``DWRF``, ``ALPHA``, ``PARQUET``, ``AVRO``, ``RCBINARY``, ``RCTEXT``, ``SEQUENCEFILE``, ``JSON``, ``TEXTFILE``, ``CSV``
+    * **Default value:** ``ORC``
+
+    This setting determines the data format for temporary tables generated by CTE materialization. The recommended value is ``PAGEFILE`` :doc:`/develop/serialized-page`, as it is the most performant,
+    since it avoids serialization and deserialization during reads and writes, allowing for direct storage of Presto pages.
+    This can also be specified on a per-query basis using the ``hive.temporary_table_storage_format`` session property.
+
+``hive.bucket-function-type-for-cte-materialization``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+    * **Type:** ``string``
+    * **Allowed values:** ``HIVE_COMPATIBLE``, ``PRESTO_NATIVE``
+    * **Default value:** ``PRESTO_NATIVE``
+
+    This setting specifies the Hash function type for cte materialization
```

Comment:
What do each of these settings mean ? Can you elaborate ?

Reply:
https://github.com/prestodb/presto/commit/cb69e899efbabf35a09003852194800309f53dad
Sample commit which added them. 


Its used https://github.com/prestodb/presto/blob/785ece64987d21d465ee66731e0d39282754e9bb/presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java#L2968
I think it determines which hashing functions to use when inserting records into the buckets.